### ISSUE TITLE
GH action for testing miner / validator

### ISSUE THREAD

**Issue by hide-on-bush-x** [masa-finance/masa-bittensor#117]
Create a github action to run automated tests of our bittensor python software.

**Comment by juanmanso** [comment]
@hide-on-bush-x from your suggestion, do you think this is dependant on Docker image generation/Docker-compose? Or is the scope of this task to develop a GitHub action that can understand if changes are to validator or miner files?

**Comment by juanmanso** [comment]
> [@hide-on-bush-x](https://github.com/hide-on-bush-x) from your suggestion, do you think this is dependant on Docker image generation/Docker-compose? Or is the scope of this task to develop a GitHub action that can understand if changes are to validator or miner files?

+1

**Comment by mudler** [comment]
Blocked until we have few tests in ( https://github.com/masa-finance/masa-bittensor/pull/248 )

**Comment by Luka-Loncar** [comment]
As @5u6r054 deploys our own validator to mainnet in automated fashion using GH actions - this ticket should get its own conclusion and applied to everything else regarding this process.

**Comment by Luka-Loncar** [comment]
In mean time it is blocked by #272

**Comment by 5u6r054** [comment]
Moving this ticket to done, as we do now have some basic code coverage happening now via a github action.

Noting there is another ticket now to extend full integration test coverage to running nodes in AWS that are registered miners and validators, and that also run masa-protocol software, which will enhance our ability make tests against the miner software (or validator software) that is continuously running, and automatically auto-updating when a new release of either repo comes out: https://github.com/masa-finance/masa-oracle-infra/issues/79
